src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/ChainPal.cs

```diff
@@ -303,25 +301,25 @@ private SafeCreateHandle GetCertsArray(IList<SafeHandle> safeHandles)
 
                     FixupStatus(cert, revocationMode, ref dwStatus);
 
-                    elements[elementIdx] = Tuple.Create(cert, dwStatus);
+                    elements[elementIdx] = (cert, dwStatus);
                 }
             }
 
             return elements;
         }
 
         private bool IsPolicyMatch(
-            Tuple<X509Certificate2, int>[] elements,
+            (X509Certificate2, int)[] elements,
             OidCollection applicationPolicy,
             OidCollection certificatePolicy)
         {
             if (applicationPolicy?.Count > 0 || certificatePolicy?.Count > 0)
             {
                 List<X509Certificate2> certsToRead = new List<X509Certificate2>();
 
-                foreach (var element in elements)
+                for (int i = 0; i < elements.Length; i++)
```

Comment:
What was wrong with foreach?

Reply:
Just an unnecessary copy of the other item when just the one item was being used.

Reply:
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAGAAkIEYUBuAWACgTyKA6AZRjAFcpsMBPFgGEo/AA4YIAcygBDMQAtBADQCsxAJxCYUDNgBm2MLIwwAznXoNCAZnIAmUkNIBvBqQ+kxvAG4mY5CikAGLQABTunlFhapraugZG/vZopNgAdhgAlADaALqkMLgwALYwmWZZkR5u9FFRADLYZhgAPLFaOnqGxqb2AHykYF1mACoQAEowsgAmpAC8pOkwAO6kTS3t6p0JPcn9YVmW1fX60KRhGRhpC6TE1DethcVlFSwN5VIYCg/YANR/Kp1eo1E4gjzDXRjSbTGYsACCMxmYSKpXKGDMOWweRYAElTCUKEcwZ4AL4ncnAjwnbzYPymQIhaDTMAKCJU+oxbbxbpJPqpK65Aqo14YoFRWrg9bNNodHmJXowAZDEbjKazW7LNYbWXcroK/aHY4cqJnWCyVkXPxQZ5ozJpdK20WVEmgk3gyEYtWwhFIlEvdF4glEyxSylRSmkoA==